src/main/scala/firrtl/Driver.scala

```diff
@@ -24,76 +24,93 @@ ANY, PROVIDED HEREUNDER IS PROVIDED "AS IS". REGENTS HAS NO OBLIGATION
 TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR
 MODIFICATIONS.
 */
-package firrtl
 
-import java.io._
-import java.nio.file.{Paths, Files}
+package firrtl
 
+import java.io.{PrintWriter, Writer, File}
 import scala.io.Source
-import scala.sys.process._
-
-import com.typesafe.scalalogging.LazyLogging
+import scala.collection.mutable
 
-import Utils._
-
-object Driver extends LazyLogging {
+object Driver {
   private val usage = """
 Usage: sbt "run-main firrtl.Driver -i <input_file> -o <output_file> -X <compiler>"
        firrtl -i <input_file> -o <output_file> -X <compiler>
 Options:
-  -X <compiler>    Specify the target language
-                   Currently supported: verilog firrtl
+  -X <compiler>    Specify the target compiler
+                   Currently supported: high low verilog
   """
-  private val defaultOptions = Map[Symbol, Any]().withDefaultValue(false)
 
-  def compile(input: String, output: String, compiler: Compiler)
+  // Compiles circuit. First parses a circuit from an input file,
+  //  executes all compiler passes, and writes result to an output
+  //  file.
+  def compile(input: String, output: String, compiler: Compiler, annotations: Seq[CircuitAnnotation] = Seq.empty)
   {
     val parsedInput = Parser.parse(input, Source.fromFile(input).getLines)
     val writerOutput = new PrintWriter(new File(output))
-    compiler.run(parsedInput, writerOutput)
+    compiler.compile(parsedInput, annotations, writerOutput)
     writerOutput.close
   }
 
-  def main(args: Array[String])
-  {
+  // Arguments specify the compiler, input file, and output file
+  def main(args: Array[String]) = {
     val arglist = args.toList
-    type OptionMap = Map[Symbol, Any]
 
+    sealed trait CompilerOption
+    case object InputFileName extends CompilerOption
+    case object OutputFileName extends CompilerOption
+    case object CompilerName extends CompilerOption
+    val defaultOptions = Map[CompilerOption, String]()
+
+    // Inline Annotation datastructure/function
+    val inlineAnnotations = mutable.HashMap[Named,Annotation]()
+    def handleInlineOption(value: String): Unit =
+       value.split('.') match {
+         case Array(module) =>
+           inlineAnnotations(ModuleName(module)) = TagAnnotation
+         case Array(module, inst) =>
+           inlineAnnotations(ComponentName(inst,ModuleName(module))) = TagAnnotation
+         case _ => throw new Exception(s"Bad inline instance/module name: $value")
+       }
+
+    type OptionMap = Map[CompilerOption, String]
     def nextOption(map: OptionMap, list: List[String]): OptionMap = {
       list match {
         case Nil => map
+        case "-inline" :: value :: tail => {
+          handleInlineOption(value)
+          nextOption(map, tail)
+        }
```

Comment:
Perhaps it's splitting hairs, but I think we should stick to the standard one dash for flaglike options, two dashes for words